src/terminal/terminal.c

```diff
@@ -413,8 +413,10 @@ guac_terminal* guac_terminal_create(guac_client* client,
 void guac_terminal_free(guac_terminal* term) {
 
     /* Close user input pipe */
-    close(term->stdin_pipe_fd[1]);
-    close(term->stdin_pipe_fd[0]);
+    if (term->stdin_pipe_fd[1] != -1)
```

Comment:
Instead of assigning -1 here, would it be better to do a fstat() call and check the return value? It seems a little bit odd to me to be manually assigning -1 to this variable above in client.c and then checking it here.

@mike-jumper Thoughts?

Reply:
`fstat()` is used get a `struct stat` for a file on a filesystem; it isn't of much use with pipe/socket descriptors.

Without weighing in on whether it makes sense to move the calls to `close`, I'd say the use of -1 to mark an unused/closed descriptor field is fairly idiomatic C.

Reply:
Fair enough.  I suggested `fstat()`, not necessarily because of any interest in the object returned, but because you could check the return value of the call for something like EBADF to see if the file descriptor is still valid/open or not.  But, if `-1` is generally accepted, then I have no problem with that.

Reply:
Thanks for the idea. If -1 is acceptable here, maybe I should also do the same in guac_terminal_free(). E.g.

void guac_terminal_free(guac_terminal* term) {

    /* Close user input pipe */
    if (term->stdin_pipe_fd[1] != -1) {
        close(term->stdin_pipe_fd[1]);
        term->stdin_pipe_fd[1] = -1;
    }
    if (term->stdin_pipe_fd[0] != -1) {
        close(term->stdin_pipe_fd[0]);
        term->stdin_pipe_fd[0] = -1;
    }

Reply:
Yeah, that would probably be fine.

Reply:
Sure. I will take a further look. Good point.